pootle/apps/pootle_format/registry.py

```diff
@@ -0,0 +1,86 @@
+
+from collections import OrderedDict
+
+from django.utils.functional import cached_property
+
+from pootle.core.delegate import format_registration
+from pootle_format.models import FileExtension, Format
+
+
+class FormatRegistry(object):
+
+    def initialize(self):
+        for filetype, info in format_registration.gather().items():
+            self.register(filetype, **info)
+
+    def register(self, name, extension, title=None, template_extension=None):
+        update = False
+        template_extension = template_extension or extension
+        try:
```

Comment:
Nit: move this `try` clause to line 29 so it is before the `if` that checks for `filetype`. The code in between is unrelated.